lib/eth/abi/event.rb

```diff
@@ -40,10 +40,20 @@ def compute_topic(interface)
       def signature(interface)
         name = interface.fetch("name")
         inputs = interface.fetch("inputs", [])
-        types = inputs.map { |i| i.fetch("type") }
+        types = inputs.map { |i| type(i) }
         "#{name}(#{types.join(",")})"
       end
 
+      def type(input)
+        if input["type"] == "tuple"
+          "(#{input["components"].map {|c| type(c) }.join(",")})"
+        elsif input["type"] == "enum"
+          "uint8"
```

Comment:
can you provide a test case for `tuple` and `enum` so I can make sure not to break it in future?